api/server/sdk/volume_migrate.go

```diff
@@ -184,6 +211,51 @@ func (s *VolumeServer) Cancel(
 	return &api.SdkCloudMigrateCancelResponse{}, nil
 }
 
+// filterStatusResponseForPermissions alters the response object to only return objects
+// that we have access to. While it seems too complicated, it minimizes the number of driver calls.
+func (s *VolumeServer) filterStatusResponseForPermissions(
+	ctx context.Context,
+	resp *api.CloudMigrateStatusResponse) (*api.CloudMigrateStatusResponse, error) {
+	var allVolIds []string
+
+	// get all volume ids to inspect
+	for _, cluster := range resp.Info {
+		for _, migrateInfo := range cluster.List {
+			allVolIds = append(allVolIds, migrateInfo.GetLocalVolumeId())
+		}
+	}
+
+	// When no vol ids are found, exit quickly
+	if len(allVolIds) == 0 {
+		return resp, nil
+	}
+
+	// get all volumes from single inspect
+	allVols, err := s.driver(ctx).Inspect(allVolIds)
+	if err != nil {
+		return nil, err
+	}
+
+	// check which volumes we have access to
+	volAccessPermitted := make(map[string]bool)
+	for _, vol := range allVols {
+		if !vol.IsPermitted(ctx, api.Ownership_Read) {
+			volAccessPermitted[vol.Id] = true
+		}
+	}
+
+	// remove volumes we don't have access to
+	for _, cluster := range resp.Info {
+		for j, migrateInfo := range cluster.List {
+			if found := volAccessPermitted[migrateInfo.GetLocalVolumeId()]; !found {
+				cluster.List = append(cluster.List[:j], cluster.List[j+1:]...)
```

Comment:
You should not adjust the same variable you are using for the for loop. It's looping is undefined. Instead just create a new list:

```go
filtered := make([]string, 0, len(allVols))
...
// if permitted:
filtered = append(filtered, migrateInfo.GetLocalVolumeId())
```

Then return the `resp` as needed